algorithms/kernel/dtrees/dtrees_model.cpp

```diff
@@ -204,7 +207,35 @@ void setNode(DecisionTreeNode& node, int featureIndex, double response)
     node.featureValueOrResponse = response;
 }
 
-services::Status addSplitNodeInternal(data_management::DataCollectionPtr& serializationData,size_t treeId, size_t parentId, size_t position, size_t featureIndex, double featureValue, size_t& res)
+void setProbabilities(const size_t treeId, const size_t nodeId, const size_t response,
+    const data_management::DataCollectionPtr probTbl, const double * const prob)
+{
+    if (probTbl.get() == nullptr)
+    {
+        return;
+    }
+    const auto treeProbaTable = (const data_management::HomogenNumericTable<double> *)(*probTbl)[treeId].get();
+    const size_t nClasses = treeProbaTable->getNumberOfRows();
+    double * const probOfTree = treeProbaTable->getArray() + nodeId * nClasses;
+    if (prob != nullptr)
+    {
+        for (size_t classIndex = 0; classIndex < nClasses; ++classIndex)
+        {
+            probOfTree[classIndex] = prob[classIndex];
+        }
+    }
+    else
+    {
+        for (size_t classIndex = 0; classIndex < nClasses; ++classIndex)
+        {
+            probOfTree[classIndex] = 0.0;
+        }
+        probOfTree[response] = 1.0;
```

Comment:
check should be added for 'response' LE than nClasses

Reply:
Check has been added in ModelBuilder::addLeafNodeInternal